src/_cffi_src/hazmat_src/padding.c

```diff
@@ -31,7 +31,7 @@ uint16_t Cryptography_check_pkcs7_padding(const uint8_t *data,
     mismatch |= Cryptography_constant_time_lt(block_len, pad_size);
 
     /* Make sure any bits set are copied to the lowest bit */
-    mismatch != mismatch >> 8;
+    mismatch |= mismatch >> 8;
```

Comment:
IMHO, both6c1d2189152c4ae764635a49a3d413f2bdb27bb8  and b021f91ae6b832b7679325dc9f1a3c58e8011af5 should be folded back into  2440514c9a4da0fb1f6b48526cfbbe6aaeeb36a7